packages/monaco/src/browser/monaco-quick-input-service.ts

```diff
@@ -15,138 +15,175 @@
  ********************************************************************************/
 /* eslint-disable @typescript-eslint/no-explicit-any */
 
-import { QuickInputService } from '@theia/core/lib/browser';
+import {
+    InputBox, InputOptions, KeybindingRegistry, PickOptions,
+    QuickInputButton, QuickInputService, QuickPick, QuickPickItem, QuickPickItemButtonEvent, QuickPickItemHighlights, QuickPickOptions, QuickPickSeparator
+} from '@theia/core/lib/browser';
 import { CancellationToken, Event } from '@theia/core/lib/common';
-import { injectable, inject, postConstruct } from '@theia/core/shared/inversify';
+import { injectable, inject } from '@theia/core/shared/inversify';
+import { MonacoResolvedKeybinding } from './monaco-resolved-keybinding';
 
 @injectable()
-export class MonacoQuickInputService extends QuickInputService implements monaco.quickInput.IQuickInputService {
+export class MonacoQuickInputService implements QuickInputService {
+    private monacoService: MonacoQuickInputImplementation;
 
     @inject(monaco.contextKeyService.ContextKeyService)
     protected readonly contextKeyService: monaco.contextKeyService.ContextKeyService;
 
-    controller: monaco.quickInput.QuickInputController;
-    quickAccess: monaco.quickInput.IQuickAccessController;
-
-    protected container: HTMLElement;
-    private quickInputList: monaco.list.List<monaco.list.IListElement>;
-
-    get backButton(): monaco.quickInput.IQuickInputButton { return this.controller.backButton; }
-    get onShow(): Event<void> { return this.controller.onShow; }
-    get onHide(): Event<void> { return this.controller.onHide; }
+    @inject(KeybindingRegistry)
+    protected readonly keybindingRegistry: KeybindingRegistry;
 
     constructor() {
-        super();
-        this.initContainer();
-        this.initController();
+        this.monacoService = new MonacoQuickInputImplementation(this.contextKeyService);
```

Comment:
Shouldn't we use inversify to inject the instance instead?

Reply:
If so you'd need to bind a factory to wrap the call to `new MonacoQuickInputImplementation`.

Reply:
The idea was that this is an implementation detail of the MonacoQuickInputService, but since we're using the same class over in MonacoEditorProvider, that is not true anyway.
The VS Code code can handle multiple instances no problem, but we don't need more than one instance, so using the one from the container seems the right thing.